### DISCUSSION TITLE
DAI 2023-2024 - My first IntelliJ + Maven project - @AlexandrePhilibert

### DISCUSSION THREAD

**Discussion by AlexandrePhilibert** [heig-vd-dai-course/heig-vd-dai-course#34] [Show and tell]
https://github.com/AlexandrePhilibert/java-intellij-idea-and-maven

**Comment by ludelafo** [comment]
Thank you!

I would add the command to download the dependencies as well so a new developer can build and run your app "from scratch".

Also, in the `pom.xml`, you only have to keep one plugin (`maven-shade-plugin`), not both (`maven-jar-plugin` + `maven-shade-plugin`). We did update the documentation about this to avoid any confusion.

Feel free to fill the poll in the GitHub Discussion regarding this chapter: https://github.com/orgs/heig-vd-dai-course/discussions/3.

You can do the same with the chapter Git, GitHub and Markdown: https://github.com/orgs/heig-vd-dai-course/discussions/2.

**Reply by AlexandrePhilibert** [reply]
Done :)

TLDR: Fixed a maven-shade-plugin warning, might not be the correct approach ?

I get this error on Linux when I run "Package as JAR file" : 

```
single version of the file is copied to the uber jar.
Usually this is not harmful and you can skip these warnings,
otherwise try to manually exclude artifacts based on
mvn dependency:tree -Ddetail=true and the above output.
See https://maven.apache.org/plugins/maven-shade-plugin/
```

![image](https://github.com/heig-vd-dai-course/heig-vd-dai-course/assets/39675639/70b30eab-c725-4b77-8382-314d7d38c960)

I followed this SO question : https://stackoverflow.com/questions/56063566/how-can-i-remove-module-info-class-warning-for-a-shaded-jar

And by adding : 

```
 <configuration>
    <filters>
        <filter>
            <artifact>*:*</artifact>
            <excludes>
                <exclude>module-info.class</exclude>
                <exclude>META-INF/**</exclude>
            </excludes>
        </filter>
    </filters>
</configuration>
```

To the `pom.xml` file, the warning seems to be gone. I don't know if this is the correct approach but it seems to fix the issue :)

**Reply by ludelafo** [reply]
Thank you very much for investigating this issue.

It seems @valentin-ricard was able to find a proper solution with his solution discussed here: https://github.com/orgs/heig-vd-dai-course/discussions/30

I will update the course material to include his solution.
